### ISSUE TITLE
Provide entity type from AggregateRootCommand

### ISSUE THREAD

**Issue by MrHash** [honeybee/honeybee#37]
AR commands could have a method to retrieve the AR type or implementor for the command. This would normalise the command builder interface and eliminate the need for unecessary type injection.

**Comment by MrHash** [comment]
The entity type is required at the moment in order to traverse attributes and sanitize the command values.

**Comment by MrHash** [comment]
Another advantage of this change would be that it's not possible to confuse a command with the wrong AR type. This happened to me once and I ended up with a event stream read error which was hard to debug.

**Comment by graste** [comment]
Shouldn't it be enough to assert that ``$command_class::getType()`` is compatible w/ the given ``$aggregate_root_type`` in the builder?

**Comment by MrHash** [comment]
Yes you could add an assertion in `AggregateRootCommand` which checks the provided ART prefix matches the command prefix. This would prevent the mismatch problem. However, the command builder uses the type so we would still need to pass it in to the builder which seems a bit unnecessary.

If the command builder had access to the ART map then we could use the command prefix to derive the type, but that's not so neat either.
